### PR TITLE
test: use `tsx` TypeScript loader

### PR DESCRIPTION
**Pull request by sebastijankuzner**
## Summary

Replace tsm with [tsx](https://www.npmjs.com/package/tsx). It support ESM modules. 

Replace [rewiremock](https://www.npmjs.com/package/rewiremock) with [esmock](https://www.npmjs.com/package/esmock). 

## Checklist

- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

File: packages/cli/source/services/source-providers/file.test.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ describe<{
 	// TODO: fix stub
 	it.skip("#install - should successfully install the plugin", async ({ source, dataPath, temporaryPath }) => {
```
Comment (oXtxNt9U) [inline packages/cli/source/services/source-providers/file.test.ts:33]: Can we now enable all the skipped tests due to stubs again?

File: packages/p2p/source/socket-server/server.test.ts [hunk 2]
```diff
@@ -72,16 +72,17 @@ describe<{ sandbox: Sandbox; server: Server }>("Server", ({ it, assert, beforeEa
 		context.server = context.sandbox.app.resolve(ServerProxy);
 	});
 
-	it("#initialize - should instantiate a new Hapi server", async ({ server }) => {
-		const spyHapiServerRegister = spy(HapiServerMock.prototype, "register");
+	it.only("#initialize - should instantiate a new Hapi server", async ({ server }) => {
```
Comment (oXtxNt9U) [inline packages/p2p/source/socket-server/server.test.ts:75]: remove .only

**Review by oXtxNt9U — APPROVED** [review]
We can cleanup in follow up PRs. Left two small remarks, but otherwise LGTM

File: packages/cli/source/services/source-providers/file.test.ts (hunk 1 above)
Comment (sebastijankuzner) [reply packages/cli/source/services/source-providers/file.test.ts:33]: Separet PR.

File: packages/p2p/source/socket-server/server.test.ts (hunk 2 above)
Comment (sebastijankuzner) [reply packages/p2p/source/socket-server/server.test.ts:75]: We are skipping all tests in server.test.ts file. I will fix that in separete PR.

**Pull request merged**